### ISSUE TITLE
uwsgi-2.0.14 build error with php plugin enabled

### ISSUE THREAD

**Issue by iron-udjin** [unbit/uwsgi#1380]
Hello,

I'm trying to build uwsgi with php plugin enabled under FreeBSD 11 use clang-3.8:
```
plugins/php/php_plugin.c:224:69: error: too few arguments to function call, expected 6, have 5
                        zend_register_string_constant(name, name_len, strval, CONST_CS, 0);
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                    ^
/usr/local/include/php/Zend/zend_constants.h:75:1: note: 'zend_register_string_constant' declared here
ZEND_API void zend_register_string_constant(const char *name, uint name_len, char *strval, int flags, int module_number TSRMLS_DC);
^
/usr/local/include/php/main/php_config.h:6:19: note: expanded from macro 'ZEND_API'
# define ZEND_API __attribute__ ((visibility("default")))
                  ^
1 error generated.
*** Error code 1
```

The same error with php-5.5 and php-5.6.

P.S: uwsgi-2.0.13 compilles fine.

**Comment by xrmx** [comment]
@gdamjan care to take a look please?

**Comment by gdamjan** [comment]
they changed the api for 7.0 it seems - let me see if I can make a patch

**Comment by gdamjan** [comment]
well,
php-5.6 (on Debian libphp5-embed  5.6.26+dfsg-0+deb8u1) has the same signature as my 7.0 on Arch
```
/usr/include/php5/Zend/zend_constants.h:ZEND_API void zend_register_string_constant(const char *name, uint name_len, char *strval, int flags, int module_number TSRMLS_DC);
```

but @iron-udjin actually has the same signature too, I don't understand the clang error then.

**Comment by gdamjan** [comment]
@iron-udjin can you try what happens if you replace the last argument in zend_register_string_constant from 0 to TSRMLS_DC (line 235 in plugins/php/php_plugin.c)

**Comment by iron-udjin** [comment]
Guys, thank you for fast respond!

@gdamjan here is result:
```
plugins/php/php_plugin.c:224:68: error: expected expression
                        zend_register_string_constant(name, name_len, strval, CONST_CS, TSRMLS_DC);
                                                                                        ^
/usr/local/include/php/TSRM/TSRM.h:166:19: note: expanded from macro 'TSRMLS_DC'
#define TSRMLS_DC       , TSRMLS_D
                        ^
plugins/php/php_plugin.c:224:68: error: expected expression
/usr/local/include/php/TSRM/TSRM.h:166:21: note: expanded from macro 'TSRMLS_DC'
#define TSRMLS_DC       , TSRMLS_D
                          ^
/usr/local/include/php/TSRM/TSRM.h:165:18: note: expanded from macro 'TSRMLS_D'
#define TSRMLS_D        void ***tsrm_ls
                        ^
2 errors generated.
*** Error code 1
```

**Comment by iron-udjin** [comment]
I've added one more argument and it compilles fine:
```
--- plugins/php/php_plugin.c.orig	2016-10-12 04:47:07.000000000 -0600
+++ plugins/php/php_plugin.c	2016-10-12 04:47:16.000000000 -0600
@@ -221,7 +221,7 @@
 			size_t name_len = equal-usl->value;
 			char *name = estrndup(usl->value, name_len);
 			char *strval = equal+1;
-			zend_register_string_constant(name, name_len, strval, CONST_CS, 0);
+			zend_register_string_constant(name, name_len, strval, CONST_CS, 0, 0);
 		}
 		usl = usl->next;
 	}
```
...but I'm not sure is it correct way or not. Can you please check?

**Comment by gdamjan** [comment]
no, it wont work on Linux (gcc). I need to find the proper macro to use in that place

**Comment by gdamjan** [comment]
@iron-udjin  try this
```
@@ -232,7 +232,7 @@ static void sapi_uwsgi_register_variables(zval *track_vars_array TSRMLS_DC)
                        size_t name_len = equal-usl->value;
                        char *name = estrndup(usl->value, name_len);
                        char *strval = equal+1;
-                       zend_register_string_constant(name, name_len, strval, CONST_CS, 0);
+                       zend_register_string_constant(name, name_len, strval, CONST_CS, 0 TSRMLS_CC);
                }
                usl = usl->next;
        }
```

**Comment by iron-udjin** [comment]
@gdamjan it works! Thank you very much!

**Comment by xrmx** [comment]
Fix pushed in both master and uwsgi-2.0 brach. Thanks again @gdamjan!
